blog/2020/07/julia-1.5-highlights.md

```diff
@@ -12,7 +12,7 @@ Let's walk through some highlights.
 
 \toc
 
-## All structs can be stack allocated
+## More inline and stack allocation
```

Comment:
This sounds like we're doing more allocation now, rather than less, which is why I changed this. Maybe "More structs can be stack allocated"?

Reply:
The inline part is important too; e.g. it changes the layout of arrays and structs. So it's not just the stack.

Reply:
Maybe "More efficient structs"? It's vague but hits the main takeaway.

Reply:
This definitely sounds like it is doing more stack allocation.

Reply:
How about "Better inlining and reduced stack allocation"?

Reply:
But it is more stack allocation?

Reply:
Duh - of course. I don't know why that sounds like a bit of a negative.

Reply:
Maybe put it as "less heap allocation"?

Reply:
I think that may be better. In general, we all look at `@time` and have programmed ourselves to think about reducing allocation. That might be why.

Similarly, I don't know, just going by the title, whether more inlining is good or bad. I mean it could be good because it reduces overhead, but bad because it generates more code, and could slow down the compiler.

How about making it explicit - `Higher performance through better inlining and stack allocation`?

Reply:
It's not code inlining, it's inline storage of structs, as in an array or a containing struct.

Reply:
Right and that is clear when you read the rest. I was trying to play the role of a casual Julia follower.

Reply:
I think that "All structs can be stack allocated" was a good title:
1. Yes "all" isn't entirely accurate, but we can explain that it's "except some really huge ones"; put an asterisk on "all" saying "except really huge ones" if you feel it needs to be 100% accurate.
2. It doesn't mention "inline" allocation, but I think people understand that if something can be stack allocated, it can also be stored inline, and if they don't get that from the title it's explained in the body.
3. Most importantly, it sounds like a win, whereas this title sounds like we're allocating more.

Reply:
How about one of

- Reduced heap allocation
- Many more structs can be stack allocated
- Struct layout and allocation optimizations

Reply:
Struct layout and allocation optimizations